crates/re_viewer_context/src/space_view/view_part_system.rs

```diff
@@ -5,10 +5,38 @@ use re_log_types::EntityPath;
 use re_types::ComponentNameSet;
 
 use crate::{
-    NamedViewSystem, SpaceViewSystemExecutionError, ViewContextCollection, ViewQuery,
-    ViewSystemName, ViewerContext,
+    NamedViewSystem, SpaceViewClassName, SpaceViewSystemExecutionError, ViewContextCollection,
+    ViewQuery, ViewSystemName, ViewerContext,
 };
 
+/// This is additional context made available to the `heuristic_filter`.
+/// It includes tree-context information such as whether certain components
+/// exist in the parent hierarchy which are better computed once than having
+/// each entity do their own tree-walk.
+#[derive(Clone, Copy, Debug)]
+pub struct HeuristicFilterContext {
+    pub class: SpaceViewClassName,
+    pub has_parent_pinhole: bool,
```

Comment:
The compute code further below suggests this is for any ancestor, not just the parent.
```suggestion
    pub has_ancestor_pinhole: bool,